ably/realtime_channel.go

```diff
@@ -190,7 +230,16 @@ func (c *RealtimeChannel) Attach(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	return res.Wait(ctx)
+
+	attachTimeout := c.client.Connection.opts.realtimeRequestTimeout()
+	timeoutCtx, cancel := c.opts().contextWithTimeout(ctx, attachTimeout)
```

Comment:
Need to check

Reply:
We do transition to `suspended` only when error is `context.DeadlineExceeded`. Can you check this one again. 

Actually, you had suggested this custom change for `channel detach` in the past and I am reusing same implementation here. So, this shouldn't be a issue.

Reply:
> We do transition to `suspended` only when error is `context.DeadlineExceeded`. Can you check this one again.

Right, so that's why we shouldn't propagate the caller-provided context to the inner code. If that context timeouts, a `context.DeadlineExceeded` will be emitted.

The fix is just this:

```go
timeoutCtx, cancel := c.opts().contextWithTimeout(context.Background(), attachTimeout)
```

> Actually, you had suggested this custom change for `channel detach` in the past and I am reusing same implementation here. So, this shouldn't be a issue.

I'm often wrong :)

Reply:
Then I am not sure, why we are passing `context` to `contextWithTImeout`, we don't need to, if it's not doing anything useful

Reply:
Because the convention is that every function that produces a context takes a context, in case you want the new context to inherit from another one. If you don't, you pass `context.Background()`.

Reply:
We can just create background context internally, rather than injecting it from outside, looking at current usecase

Reply:
Yes, we could. But then that makes the function less flexible. Anyway, feel free to change it if you feel strongly about it.